plugins/woocommerce/src/Blocks/BlockTemplatesController.php

```diff
@@ -355,16 +332,26 @@ public function add_block_templates( $query_result, $query, $template_type ) {
 				continue;
 			}
 
-			$is_not_custom   = false === array_search(
+			$possible_template_ids = [
 				$theme_slug . '//' . $template_file->slug,
-				array_column( $query_result, 'id' ),
-				true
-			);
+				$theme_slug . '//' . BlockTemplateUtils::DIRECTORY_NAMES['TEMPLATE_PARTS'] . '/' . $template_file->slug,
+				$theme_slug . '//' . BlockTemplateUtils::DIRECTORY_NAMES['DEPRECATED_TEMPLATE_PARTS'] . '/' . $template_file->slug,
```

Comment:
Template parts in classic themes that support template parts seemed to have a different ID than expected, that's the reason of this change.